src/compose.ts

```diff
@@ -346,3 +381,175 @@ function normalizeKeyValuePairs(
 		.fromPairs()
 		.value();
 }
+
+/**
+ * Reads a normalized composition object and expands context of composition
+ * Context is related to the composition in terms of env_file tag in composition
+ * env_file tag values (file paths) are normalized and validated.
+ *
+ * Caches environment files for all services in the composition and transfers all env_file
+ * environment vars into the environment tag of the composition.
+ *
+ * Drops env_file propertie from composition to indicate that the expand has taken place
+ *
+ * @param normalizedComposition normalized composition object
+ * @param fileResolverCb callback from caller that returns a readStream for a given filePath string
+ * @returns Composition with expanded context
+ */
```

Comment:
This function is file-private and fairly straightforward. A docstring for these is good to have but don’t feel obliged to provide one if it’s not adding useful information that can’t be easily deduced by reading the code as they tend to quickly get out of sync with each other and become noise.

Reply:
Agreed, I'll delete the docs as, as you say, they tent to outdate.